sdk/core/azure-core/src/http/user_agent.cpp

```diff
@@ -133,10 +133,10 @@ std::string GetOSVersion()
 
 std::string TrimString(std::string s)
 {
-  auto const isSpace = [](int c) { return !std::isspace(c); };
+  auto const isNotSpace = [](char c) { return !std::isspace(c, std::locale::classic()); };
 
-  s.erase(s.begin(), std::find_if(s.begin(), s.end(), isSpace));
-  s.erase(std::find_if(s.rbegin(), s.rend(), isSpace).base(), s.end());
+  s.erase(s.begin(), std::find_if(s.begin(), s.end(), isNotSpace));
+  s.erase(std::find_if(s.rbegin(), s.rend(), isNotSpace).base(), s.end());
```

Comment:
The double negative in `isNotSpace` is a bit confusing, especially combined with what `find_if` does, which is:
> An iterator to the first element in the range for which pred does not return false.

Would using an `isSpace` function be more clear, logically?

Reply:
To be clear, I did not touch the logic, but I updated the name to actually correspond to what the function does, which before was the exact opposite of the name :)

I agree, using "actual" `isSpace` could have been better, however for std::erase we couldn't put it with a negation operator, i.e. unfortunately we can't write `find_if(..., ..., !isSpace)`, so "isNotSpace" is needed.

Look at it this way, it is not unreadable, and it is not a double negation, although normally functions like "isNotSomething()" are not common.
It is readable as this: "erase from front of the string to the first character from beginning that is NOT space;" then next line: "erase from the back of the string (rbegin) to the first character from the end that is NOT space". In speech it sounds more normal, as it is more normal for us to operate with wording like "first non-space character", as we don't seem to have a better word for it except calling such characters "non-space".